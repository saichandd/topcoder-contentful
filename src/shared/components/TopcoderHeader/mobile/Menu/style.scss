@import '~styles/mixins';

.menu {
  background: $tc-gray-90;
  // height: 58px;
  margin-top: -55px;
  position: relative;
  width: 100%;
  top: 0;
  z-index: 30;
}

.separator {
  background: $tc-gray-40;
  height: 1px;
  margin: 0 12px;
}

.tc-logo {
  left: 12px;
  position: absolute;
}

.title {
  color: $tc-white;
  padding: 12px;
  text-align: center;
}

.x-cross {
  color: $tc-white;
  cursor: pointer;
  font: 100 48px "Sofia Pro", sans-serif;
  position: absolute;
  right: 12px;
  top: -18px;
}
