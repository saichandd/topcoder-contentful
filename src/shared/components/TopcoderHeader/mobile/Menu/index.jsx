import PT from 'prop-types';
import React from 'react';

import Auth from '../../Auth';
import LogoTopcoder from '../../../../../assets/images/logo_topcoder.svg';
import Search from '../Search';
import SubMenu, { SUB_MENU_SHAPE } from '../SubMenu';
import UserMenu from '../UserMenu';

import './style.scss';

export default function Menu({
  close,
  mainMenu,
  profile,
  userMenu,
}) {
  return (
    <div styleName="menu">
      <div styleName="title">
        <LogoTopcoder styleName="tc-logo" />
        [ topcoder ]
        <span
          onClick={close}
          onKeyPress={close}
          role="button"
          styleName="x-cross"
          tabIndex={0}
        >
&times;
        </span>
      </div>
      <Search />
      {
        profile
          ? <UserMenu menu={userMenu} profile={profile} />
          : <Auth column />
      }
      <div styleName="separator" />
      {mainMenu.map(item => <SubMenu key={item.title} subMenu={item} />)}
    </div>
  );
}

Menu.defaultProps = {
  profile: null,
  userMenu: null,
};

Menu.propTypes = {
  close: PT.func.isRequired,
  mainMenu: PT.arrayOf(SUB_MENU_SHAPE).isRequired,
  profile: PT.shape({}),
  userMenu: SUB_MENU_SHAPE,
};
