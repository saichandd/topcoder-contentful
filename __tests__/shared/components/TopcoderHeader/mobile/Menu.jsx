import _ from 'lodash';
import React from 'react';
import R from 'react-test-renderer/shallow';
import Menu from 'components/TopcoderHeader/mobile/Menu';

const renderer = new R();

test('Matches snapshot', () => {
  renderer.render((
    <Menu
      close={_.noop}
      mainMenu={[{
        title: 'Menu title',
        items: [{
          icon: <div />,
          link: '/LINK',
          title: 'Item title',
        }],
      }]}
    />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
