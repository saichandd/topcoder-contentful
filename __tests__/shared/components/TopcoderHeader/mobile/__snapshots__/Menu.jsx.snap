// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="src-shared-components-TopcoderHeader-mobile-Menu-___style__menu___3UwIt"
>
  <div
    className="src-shared-components-TopcoderHeader-mobile-Menu-___style__title___2L9ZL"
  >
    <LogoTopcoder
      className="src-shared-components-TopcoderHeader-mobile-Menu-___style__tc-logo___2ebQz"
      height="18"
      viewBox="0 0 51 18"
      width="51"
      xmlns="http://www.w3.org/2000/svg"
    />
    [ topcoder ]
    <span
      className="src-shared-components-TopcoderHeader-mobile-Menu-___style__x-cross___th5Kv"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      Ã—
    </span>
  </div>
  <Search />
  <Auth
    column={true}
  />
  <div
    className="src-shared-components-TopcoderHeader-mobile-Menu-___style__separator___3LchD"
  />
  <SubMenu
    hideTitle={false}
    subMenu={
      Object {
        "items": Array [
          Object {
            "icon": <div />,
            "link": "/LINK",
            "title": "Item title",
          },
        ],
        "title": "Menu title",
      }
    }
  />
</div>
`;
